libpod/runtime_ctr.go

```diff
@@ -667,3 +676,58 @@ func (r *Runtime) GetLatestContainer() (*Container, error) {
 	}
 	return ctrs[lastCreatedIndex], nil
 }
+
+//GetStorageContainers returns a list of containers from containers/storage that
+//are not currently known to Podmman.
+func (r *Runtime) GetStorageContainers(ctrs []*Container) ([]*Container, error) {
+	var retCtrs []*Container
+
+	// We only have a store when doing all, storage or size, so skip this otherwise.
+	if r.store == nil {
+		return retCtrs, nil
+	}
+
+	// Make a map of the libpod containers.  If we run across them from
+	// c/storage, we won't grab them in the processing below.
+	libpodCtrs := make(map[string]bool)
+	for _, ctr := range ctrs {
+		libpodCtrs[ctr.ID()] = true
+	}
+
+	storeContainers, err2 := r.store.Containers()
+	if err2 != nil {
+		return ctrs, errors.Wrapf(err2, "error reading list of all containers")
+	}
+	for _, container := range storeContainers {
+		// No need to handle libpod containers here.
+		if _, foundCtr := libpodCtrs[container.ID]; foundCtr {
+			continue
+		}
+		newCtr := new(Container)
+		newCtr.config = new(ContainerConfig)
```

Comment:
I would not expect building containers like this to work; soe of this, like lock allocation, has some definite undesirable side-effects.

We should confine the storage logic to cmd/podman, and ideally keep it separate from the rest of `ps`. If a storage container hits Libpod, I do not expect good things to happen.